lisa/features/disks.py

```diff
@@ -79,20 +81,41 @@ def get_os_boot_partition(self) -> Optional[PartitionInfo]:
         for partition in partition_info:
             if partition.mount_point.startswith("/boot"):
                 boot_partition = partition
+                if isinstance(self._node.os, BSD):
+                    # Get the device name from the GPT since they are abstracted
+                    # Ex. /boot is mounted on /gpt/efiesp
+                    # This is the output of gpart show.
+                    # Name           Status  Components
+                    # gpt/efiesp     N/A     da0p1
+                    # gpt/rootfs     N/A     da0p2
+                    _get_device_from_gpt_bsd_regex = re.compile(
+                        r"\n?" + re.escape(boot_partition.disk) + r"\s*\S*\s*(\S*)"
+                    )
+                    cmd = "glabel status"
+                    output = self._node.execute(cmd).stdout
+                    dev = get_matched_str(output, _get_device_from_gpt_bsd_regex)
+                    boot_partition.disk = dev
                 break
         return boot_partition
 
     # Get disk controller type from the VM by checking the boot partition
     def get_os_disk_controller_type(self) -> schema.DiskControllerType:
         boot_partition = self.get_os_boot_partition()
         assert boot_partition, "'boot_partition' must not be 'None'"
-
-        if boot_partition.disk.startswith("nvme"):
-            os_disk_controller_type = schema.DiskControllerType.NVME
-        elif boot_partition.disk.startswith("sd"):
-            os_disk_controller_type = schema.DiskControllerType.SCSI
+        if "FreeBSD" in self._node.os.name:
```

Comment:
Use `isinstance` as above.

Reply:
Ah I missed the second one. Thanks for pointing it out.